src/Framework/Sdk/SdkResultItem.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// The value of an item and any associated metadata to be added by an SDK resolver.  See <see cref="SdkResult.ItemsToAdd"/>
+    /// </summary>
+    public class SdkResultItem
+    {
+        public string ItemSpec { get; set; }
+        public Dictionary<string, string> Metadata { get;}
+
+        public SdkResultItem()
+        {
+            Metadata = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="SdkResultItem"/>
+        /// </summary>
+        /// <param name="itemSpec">The value (itemspec) for the item</param>
+        /// <param name="metadata">A dictionary of item metadata.  This should be created with <see cref="StringComparer.OrdinalIgnoreCase"/> for the comparer.</param>
+        public SdkResultItem(string itemSpec, Dictionary<string, string> metadata)
+        {
+            ItemSpec = itemSpec;
+            Metadata = metadata;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is SdkResultItem item &&
+                   ItemSpec == item.ItemSpec &&
+                   Metadata?.Count == item.Metadata?.Count)
+            {
+                if (Metadata != null)
+                {
+                    foreach (var kvp in Metadata)
+                    {
+                        if (item.Metadata[kvp.Key] != kvp.Value)
+                        {
+                            return false;
+                        }
+                    }
+                }
+
+                return true;
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = -849885975;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(ItemSpec);
+
+            if (Metadata != null)
+            {
+                foreach (var kvp in Metadata)
+                {
+                    hashCode = hashCode * 1521134295 + kvp.Key.GetHashCode();
```

Comment:
This is negative everywhere else. Is the positive here intentional?

Also, this copies elsewhere but I'm just noticing it now: should this be an explicitly unchecked operation? Couldn't this cause overflow?

Reply:
I used the VS code helper to generate these methods, and then modified as appropriate for loops, etc.  So I think the missing negative was a copy/paste error.  I don't know whether they should be explicitly unchecked, but if they should be, I would expect the VS code helper to include that.